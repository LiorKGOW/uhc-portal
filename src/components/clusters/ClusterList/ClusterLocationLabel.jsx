// ClusterLocationLabel shows the location of the cluster in the form of
// "Cloud Provider (human readable location name)".

import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { cloudProviderActions } from '../../../redux/actions/cloudProviderActions';

const awsRegions = {
  'us-east-2': 'US East (Ohio)',
  'us-east-1': 'US East (N. Virginia)',
  'us-west-1': 'US West (N. California)',
  'us-west-2': 'US West (Oregon)',
  'ap-south-1': 'Asia Pacific (Mumbai)',
  'ap-northeast-3': 'Asia Pacific (Osaka-Local)',
  'ap-northeast-2': 'Asia Pacific (Seoul)',
  'ap-southeast-1': 'Asia Pacific (Singapore)',
  'ap-southeast-2': 'Asia Pacific (Sydney)',
  'ap-northeast-1': 'Asia Pacific (Tokyo)',
  'ca-central-1': 'Canada (Central)',
  'cn-north-1': 'China (Beijing)',
  'cn-northwest-1': 'China (Ningxia)',
  'eu-central-1': 'EU (Frankfurt)',
  'eu-west-1': 'EU (Ireland)',
  'eu-west-2': 'EU (London)',
  'eu-west-3': 'EU (Paris)',
  'eu-north-1': 'EU (Stockholm)',
  'sa-east-1': 'South America (SÃ£o Paulo)',
  'us-gov-east-1': 'AWS GovCloud (US-East)',
  'us-gov-west-1': 'AWS GovCloud (US)',
};


class ClusterLocationLabel extends React.Component {
  componentDidMount() {
    const { getCloudProviders, cloudProviders } = this.props;
    if (!cloudProviders.pending && !cloudProviders.error && !cloudProviders.fulfilled) {
      getCloudProviders();
    }
  }

  render() {
    const { regionID, cloudProviderID, cloudProviders } = this.props;
    if (cloudProviderID === 'aws' && awsRegions[regionID]) {
      return (
        <React.Fragment>
          AWS
          {' '}
          {awsRegions[regionID]}
        </React.Fragment>
      );
    }
    if (cloudProviders.fulfilled
        && cloudProviders.providers[cloudProviderID]
        && cloudProviders.providers[cloudProviderID].regions
        && cloudProviders.providers[cloudProviderID].regions[regionID]) {
      return (
        <React.Fragment>
          {cloudProviders.providers[cloudProviderID].name.toUpperCase()}
          {` ${cloudProviders.providers[cloudProviderID].regions[regionID].display_name}`}
        </React.Fragment>
      );
    }
    return (
      <React.Fragment>
        {cloudProviderID.toUpperCase()}
        {` (${regionID})`}
      </React.Fragment>
    );
  }
}

ClusterLocationLabel.propTypes = {
  regionID: PropTypes.string.isRequired,
  cloudProviderID: PropTypes.string.isRequired,
  getCloudProviders: PropTypes.func.isRequired,
  cloudProviders: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  cloudProviders: state.cloudProviders.cloudProviders,
});

const mapDispatchToProps = {
  getCloudProviders: cloudProviderActions.getCloudProviders,
};

export default connect(mapStateToProps, mapDispatchToProps)(ClusterLocationLabel);
