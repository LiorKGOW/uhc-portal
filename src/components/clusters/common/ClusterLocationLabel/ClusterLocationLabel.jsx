// ClusterLocationLabel shows the location of the cluster in the form of
// "Cloud Provider (location name)".

import React from 'react';
import PropTypes from 'prop-types';

class ClusterLocationLabel extends React.Component {
  componentDidMount() {
    const { getCloudProviders, cloudProviders } = this.props;
    if (!cloudProviders.pending && !cloudProviders.error && !cloudProviders.fulfilled) {
      getCloudProviders();
    }
  }

  render() {
    const { regionID, cloudProviderID, cloudProviders } = this.props;
    const provider = cloudProviders.fulfilled && cloudProviders.providers[cloudProviderID]
      ? cloudProviders.providers[cloudProviderID].display_name
      : cloudProviderID.toUpperCase();

    return (
      <>
        {provider}
        {regionID !== 'N/A' && ` (${regionID})`}
      </>
    );
  }
}

ClusterLocationLabel.propTypes = {
  regionID: PropTypes.string.isRequired, // parent component passes N/A when unknown
  cloudProviderID: PropTypes.string.isRequired, // parent component passes N/A when unknown
  getCloudProviders: PropTypes.func.isRequired,
  cloudProviders: PropTypes.object.isRequired,
};

export default ClusterLocationLabel;
