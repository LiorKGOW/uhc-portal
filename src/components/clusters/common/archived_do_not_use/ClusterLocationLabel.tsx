// ClusterLocationLabel shows the location of the cluster in the form of
// "Cloud Provider (location name)".

import React from 'react';
import { useDispatch } from 'react-redux';

import { getCloudProviders } from '~/redux/actions/cloudProviderActions';
import { useGlobalState } from '~/redux/hooks';

export type ClusterLocationLabelProps = {
  regionID: string;
  cloudProviderID: string;
};
const ClusterLocationLabel = ({ regionID, cloudProviderID }: ClusterLocationLabelProps) => {
  const dispatch = useDispatch();
  const cloudProviders = useGlobalState((state) => state.cloudProviders);

  React.useEffect(() => {
    if (!cloudProviders.pending && !cloudProviders.error && !cloudProviders.fulfilled) {
      dispatch(getCloudProviders());
    }
  }, [dispatch, cloudProviders]);

  return (
    <>
      {cloudProviders.fulfilled && cloudProviders.providers[cloudProviderID]
        ? cloudProviders.providers[cloudProviderID].display_name
        : cloudProviderID.toUpperCase()}
      {regionID !== 'N/A' && ` (${regionID})`}
    </>
  );
};

export default ClusterLocationLabel;
