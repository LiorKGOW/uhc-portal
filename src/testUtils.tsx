import React from 'react';
import { Provider } from 'react-redux';
import { createStore, Store } from 'redux';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import { toHaveNoViolations, axe } from 'jest-axe';

import { createBrowserHistory } from 'history';
import { store } from './redux/store';
import { reduxReducers } from './redux/reducers';
import * as restrictedEnv from './restrictedEnv';

const history = createBrowserHistory();

export const TestWrapper = ({
  children,
  initialStore,
}: {
  children: React.ReactNode;
  initialStore?: Store;
}) => <Provider store={initialStore || store}>{children}</Provider>;

const renderWithState = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  initialState?: any, // setting to any because only a partial state structure may be sent
) => {
  const newStore = initialState ? createStore(reduxReducers(history), initialState) : store;

  const Wrapper = ({ children }: { children: React.ReactNode }) => (
    <TestWrapper initialStore={newStore}>{children}</TestWrapper>
  );
  return { ...render(ui, { wrapper: Wrapper, ...options }), user: userEvent.setup() };
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// Type not exported in the library
export type UserEventType = {
  type: (
    field: HTMLElement,
    typeValue: string,
    { initialSelectionStart }: { initialSelectionStart: number },
  ) => void;
  clear: (field: HTMLElement) => void;
};

export { renderWithState as render };

/* ***** Items outside of React Test Library ************ */
expect.extend(toHaveNoViolations);

export const checkAccessibility = async (container: HTMLElement | string) => {
  const results = await axe(container);
  expect(results).toHaveNoViolations();
};

export const insightsMock = () => {
  global.insights = {
    chrome: {
      ...global.insights.chrome,
      on: () => () => {},
      appNavClick: () => {},
      auth: {
        getUser: () => Promise.resolve({ data: {} }),
        getToken: () => Promise.resolve(),
        getOfflineToken: () => Promise.resolve({ data: { refresh_token: 'hello' } }),
      },
    },
  };
};

export const mockRestrictedEnv = () => {
  const mock = jest.spyOn(restrictedEnv, 'isRestrictedEnv');
  mock.mockReturnValue(false);
  return mock;
};
