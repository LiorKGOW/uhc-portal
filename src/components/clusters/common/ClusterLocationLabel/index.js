import { connect } from 'react-redux';
import { cloudProviderActions } from '../../../../redux/actions/cloudProviderActions';
import ClusterLocationLabel from './ClusterLocationLabel';

const mapStateToProps = state => ({
  cloudProviders: state.cloudProviders,
});

const mapDispatchToProps = {
  getCloudProviders: cloudProviderActions.getCloudProviders,
};

export default connect(mapStateToProps, mapDispatchToProps)(ClusterLocationLabel);
