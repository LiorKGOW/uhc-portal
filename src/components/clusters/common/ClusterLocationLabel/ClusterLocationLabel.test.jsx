import React from 'react';
import { render } from '~/testUtils';
import ClusterLocationLabel from './ClusterLocationLabel';

describe('<ClusterLocationLabel />', () => {
  const getCloudProviders = jest.fn();

  const defaultProps = {
    getCloudProviders,
    cloudProviderID: 'aws',
    regionID: 'us-east-1',
    cloudProviders: {
      pending: false,
      fulfilled: false,
      error: false,
      providers: {},
    },
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('fetches cloud providers on render', () => {
    expect(getCloudProviders).not.toBeCalled();
    render(<ClusterLocationLabel {...defaultProps} />);
    expect(getCloudProviders).toBeCalled();
  });

  it("displays cloud provider and region when there's no provider data available", () => {
    expect(defaultProps.cloudProviders.providers).toEqual({});
    const { container } = render(<ClusterLocationLabel {...defaultProps} />);

    expect(container.textContent).toEqual('AWS (us-east-1)');
  });

  it('does not display region  when region is not known', () => {
    const newProps = { ...defaultProps, cloudProviderID: 'gcp', regionID: 'N/A' };
    const { container } = render(<ClusterLocationLabel {...newProps} />);

    expect(container.textContent).toEqual('GCP');
  });

  it('displays content from providers when providers have been fetched', () => {
    const newProps = {
      ...defaultProps,
      cloudProviderID: 'baremetal',
      regionID: 'some-region',
      cloudProviders: { fulfilled: true, providers: { baremetal: { display_name: 'Bare Metal' } } },
    };

    const { container } = render(<ClusterLocationLabel {...newProps} />);

    expect(container.textContent).toEqual('Bare Metal (some-region)');
  });
});
