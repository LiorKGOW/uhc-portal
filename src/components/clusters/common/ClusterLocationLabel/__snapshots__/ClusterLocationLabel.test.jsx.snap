// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClusterLocationLabel /> renders correctly when providers have been fetched 1`] = `
<Fragment>
  Bare Metal
   (some-region)
</Fragment>
`;

exports[`<ClusterLocationLabel /> renders correctly when region is not known 1`] = `
<Fragment>
  GCP
</Fragment>
`;

exports[`<ClusterLocationLabel /> renders correctly when there's no provider data available 1`] = `
<Fragment>
  AWS
   (us-east-1)
</Fragment>
`;
